addons/common/functions/fnc_statusEffect_set.sqf

```diff
@@ -62,9 +61,13 @@ if (_set isEqualTo (_effectBoolArray select _statusIndex)) exitWith {
 
 TRACE_2("Setting to new value",_set,_effectBoolArray select _statusIndex);
 _effectBoolArray set [_statusIndex, _set];
-_effectNumber = _effectBoolArray call FUNC(toBitmask); //Convert array back to number
+private _newEffectNumber = _effectBoolArray call FUNC(toBitmask); //Convert array back to number
 
-TRACE_2("Saving globaly",_effectVarName,_effectNumber);
-_object setVariable [_effectVarName, _effectNumber, true];
+TRACE_2("Saving globally",_effectVarName,_newEffectNumber);
+_object setVariable [_effectVarName, _newEffectNumber, true];
 
-[_object, _effectName] call FUNC(statusEffect_sendEffects);
+if (_effectNumber == 0 || {_newEffectNumber == 0}) then {
```

Comment:
I'm 99% sure this is a safe optimzation
All of our use is `_set > 0` or `_set == 0`
so going from forceWalk _set=1 [missionSet] to _set=3 [missionSet + dragging] would be the same

In theory someone could be doing something weird and handling the effect based on the number of sources
But I doubt it